js/try.js

```diff
@@ -102,7 +157,24 @@ const attachToForm = () => {
 
   const linkInput = document.querySelector('input#link')
   const linkForm = document.querySelector('form')
-  const retrieveButton = document.querySelector('form button')
+  const retrieveButton = document.querySelector('#retrieve');
+  const downloadButton = document.querySelector('#download');
+
+  downloadButton.addEventListener("click", async (e) => {
+    const identifier = getIdentifier(linkInput.value);
```

Comment:
Add `e.preventDefault()` here, otherwise pressing the 'Download' button calls both this and the next event listener.